web/src/pages/Proto/List.tsx

```diff
@@ -0,0 +1,170 @@
+
+  const refreshTable = () => {
+    ref.current?.reload();
+  };
+
+  const savePageList = (page = 1, pageSize = 10) => {
+    history.replace(`/proto/list?page=${page}&pageSize=${pageSize}`);
+  };
+
+  useEffect(() => {
+    const { page = 1, pageSize = 10 } = querystring.parse(window.location.search);
+    setPaginationConfig({ pageSize: Number(pageSize), current: Number(page) });
+  }, [window.location.search]);
+
+  const showDrawer = (data: ProtoModule.ProtoData, mode: ProtoModule.EditMode) => {
+    setDrawerVisible(true);
+    setProtoData(data);
+    setEditMode(mode);
+  };
+
+  const columns: ProColumns<ProtoModule.ResponseBody>[] = [
+    {
+      title: formatMessage({ id: 'component.global.id' }),
+      hideInSearch: true,
+      dataIndex: 'id',
+      width: 100,
+    },
+    {
+      title: formatMessage({ id: 'page.proto.desc' }),
+      dataIndex: 'desc',
+      ellipsis: true,
+      width: 200,
+    },
+    {
+      title: formatMessage({ id: 'page.proto.content' }),
+      hideInSearch: true,
+      dataIndex: 'content',
+      ellipsis: true,
+    },
+    {
+      title: formatMessage({ id: 'component.global.updateTime' }),
+      dataIndex: 'update_time',
+      hideInSearch: true,
+      render: (text) => timestampToLocaleString(text as number),
+      width: 200,
+    },
+    {
+      title: formatMessage({ id: 'component.global.operation' }),
+      valueType: 'option',
+      fixed: 'right',
+      hideInSearch: true,
+      render: (_, record) => (
+        <>
+          <Space align="baseline">
+            <Button
+              type="primary"
+              onClick={() =>
+                showDrawer({ id: record.id, content: record.content, desc: record.desc }, 'update')
+              }
+            >
+              {formatMessage({ id: 'component.global.edit' })}
+            </Button>
+            <Popconfirm
+              title={formatMessage({ id: 'page.upstream.list.confirm.delete' })}
+              okText={formatMessage({ id: 'page.upstream.list.confirm' })}
+              cancelText={formatMessage({ id: 'page.upstream.list.cancel' })}
+              onConfirm={() => {
+                remove(record.id!).then(() => {
+                  notification.success({
+                    message: formatMessage({ id: 'page.upstream.list.delete.successfully' }),
+                  });
+                  /* eslint-disable no-unused-expressions */
+                  ref.current?.reload();
+                });
+              }}
+            >
+              <Button type="primary" danger>
+                {formatMessage({ id: 'page.upstream.list.delete' })}
+              </Button>
+            </Popconfirm>
+          </Space>
+        </>
+      ),
+    },
+  ];
+
+  return (
+    <>
+      <ProtoDrawer
+        protoData={protoData as ProtoModule.ProtoData}
+        setProtoData={setProtoData}
+        visible={drawerVisible}
+        setVisible={setDrawerVisible}
```

Comment:
we can use a state in this component control visible like `visible && ProtoDrawer`, this way avoided passing many props

Reply:
There is a coupling between these two components.